### ISSUE TITLE
Beautify does not

### ISSUE THREAD

**Issue by justintocci** [workflowproducts/postage#183]
CREATE OR REPLACE VIEW jps.qry_schedule_all_color AS 
SELECT tjobs_print_details.id, tjobs_print_details.itemstatusid AS tjobs_print_details_status_id, tjobs.id AS jobno, tjobs.entrydate, tjobs_print_details.operduedate, CASE WHEN tjobs_print_details.printertype = 0 THEN

'Océ Only'::text WHEN tjobs_print_details.printertype = 2 THEN

	'Ryobi'::text WHEN tjobs_print_details.printertype = ANY (ARRAY[1, 6, 7, 8, 9, 11]) THEN

		'Xerox'::text WHEN tjobs_print_details.printertype = 12 THEN

			'GTO'::text WHEN tjobs_print_details.printertype = 13 THEN

				'Xerox Color'::text WHEN tjobs_print_details.printertype = 14 THEN

					'Indigo'::text WHEN tjobs_print_details.printertype = 15 THEN

						'Epson 6000'::text WHEN tjobs_print_details.printertype = 19 THEN

							'Flatbed'::text WHEN tjobs_print_details.printertype IS NULL THEN

								'None'::text ELSE tjobs_print_details.printertype::text ||- '-UNKNOWN'::text END

**Comment by justintocci** [comment]
```
GRANT SELECT, UPDATE, INSERT, DELETE, TRUNCATE, REFERENCES, TRIGGER ON TABLE jps.qry_schedule_all_color TO postgres;

										
GRANT SELECT,
										UPDATE ON TABLE jps.qry_schedule_all_color TO msaccess_g;

										
GRANT
										SELECT ON TABLE jps.qry_schedule_all_color TO trusted_g;

										CREATE TRIGGER trg_instead_qry_schedule_all_color INSTEAD OF
										UPDATE ON jps.qry_schedule_all_color FOR EACH ROW EXECUTE PROCEDURE jps.instead_qry_schedule_all_color() ;
```

**Comment by justintocci** [comment]
However, if these things can be fixed, this will be my favorite button. It indents VIEWS with subqueries very well and makes them easy to read.

**Comment by josephtocci** [comment]
THEN indents because of IF THEN, I didn't think of CASE WHEN. The solution is to increment a variable when CASE is detected, then until END is detected. Then only indent the THENs when to variable is 0;

I have no idea what's happening to the GRANT statements.

I did minimal testing for the beautify because it was low priority. It did work well for the SELECT statements I tested because that's when you need a beautifier.

**Comment by justintocci** [comment]
I don't mean to nitpick, but this:
````
LEFT JOIN (
    SELECT tjobs_binding_details.job_id, string_agg(tbinding_categories.describes::text, ','::text) AS bind
    FROM jps.tjobs_binding_details
    LEFT JOIN jps.tbinding_categories ON tbinding_categories.id = tjobs_binding_details.bindingid
    GROUP BY tjobs_binding_details.job_id) jbd ON jbd.job_id = tjobs.id
````

would look even better as:
````
LEFT JOIN (
    SELECT tjobs_binding_details.job_id, string_agg(tbinding_categories.describes::text, ','::text) AS bind
    FROM jps.tjobs_binding_details
    LEFT JOIN jps.tbinding_categories ON tbinding_categories.id = tjobs_binding_details.bindingid
    GROUP BY tjobs_binding_details.job_id
    ) jbd ON jbd.job_id = tjobs.id
````

**Comment by justintocci** [comment]
Oh my gosh, it's deleting characters!!

END,','::text

becomes:

END
     
    ','::text

**Comment by josephtocci** [comment]
I figured out how it was deleting characters, fixed.

**Comment by josephtocci** [comment]
Beautify has been completely rewritten from scratch. I uses @crosstocci sql split code he brought over from C to make sure we don't screw up string literals. This also means we don't need underscore.js which I had @nunziotocci remove. @crosstocci is adding some keywords to my code, once he is done he will close the issue and further feature requests for the beautify should be done in new issues.
